pytorch_lightning/trainer/trainer.py

```diff
@@ -1494,6 +1500,42 @@ def call_hook(
 
         return output
 
+    def _profile_and_call(self, obj: object, hook_name: str, *args: Any, **kwargs: Any) -> Optional[Any]:
+        fn = getattr(obj, hook_name)
+        with self.profiler.profile(f"{obj.__class__.__name__}.{hook_name}"):
+            return fn(*args, **kwargs)
+
+    def _call_hook(
+        self,
+        obj: object,
+        hook_name: str,
+        *args: Any,
+        pl_module: Optional["pl.LightningModule"] = None,
+        **kwargs: Any,
+    ) -> Optional[Any]:
+        pl_module = pl_module or self.lightning_module
+
+        if pl_module is not None:
+            prev_fx_name = pl_module._current_fx_name
+            pl_module._current_fx_name = hook_name
```

Comment:
this means we're setting the hook name even for callback hooks, not lightning module hooks. is that correct?

Reply:
I don't think so. it seems to me that `_current_fx_name` is just for the LM hooks so if we go with the three method approach, we can just include this logic directly in `_call_lightning_module_hook`?  

I am also a bit confused because we sometimes just call the LM hook directly, instead of calling `call_hook`:
https://github.com/PyTorchLightning/pytorch-lightning/blob/823bfa6f8a16a8ff2f93c8d5893549424e5b471b/pytorch_lightning/loops/dataloader/evaluation_loop.py#L235

Is this intentional or is this a mistake?